cantools/database/diagnostics/data.py

```diff
@@ -57,6 +60,23 @@ def __init__(self,
         self.is_float: bool = False
         self.is_signed: bool = False
 
+    def get_scaling_offset(self, raw_val):
+        def convert(v, o, f):
+            return (v - o) / f
+
+        if not isinstance(self.minimum, list):
```

Comment:
For piecewise linear types it is actually necessary, so if it's not there we can be sure that we don't need to deal with the rest and can just directly return the offset and scale

However, once I've refactored the data class to use a separate member for defining the piecewise segments this should anyways get clearer

Reply:
i was rather thinking that instead of checking if `self.minimum` is a list, it would probably slightly clearer to use `self.offset` or `self.scale`... (not a bit deal, though)